### PR TITLE
ADD: initial AWS testing resources CDK

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
[Describe what this change achieves]
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: testing/aws-testing-cdk/lib/aws-secrets-manager/SecretsManagerStack.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  readonly openSearchSinkSecrets : Secret;
+
+  constructor(scope: Construct, id: string, props: SecretsManagerStackProps) {
+    super(scope, id, props);
+
+    this.openSearchSinkSecrets = new Secret(this, 'OpenSearchSinkSecrets', {
+      secretName: 'opensearch-sink-basic-credentials',
+      secretObjectValue: {
+        // TODO: pass in secret values as args?
+        username: SecretValue.unsafePlainText('admin'),
```
Comment (dlvenable) [inline testing/aws-testing-cdk/lib/aws-secrets-manager/SecretsManagerStack.ts:20]: Maybe use a value which is less likely to be used elsewhere. For example: `data-prepper-admin`.

**Review by dlvenable — CHANGES_REQUESTED** [review]
This is looking very nice. Thanks for starting this initiative!

File: testing/aws-testing-cdk/lib/aws-secrets-manager/SecretsManagerStack.ts (hunk 1 above)
Comment (chenqi0805) [reply testing/aws-testing-cdk/lib/aws-secrets-manager/SecretsManagerStack.ts:20]: Not sure how easy it is to reset the username/password with the opensearch docker image. Will leave it for future change.

Comment (dlvenable) [reply testing/aws-testing-cdk/lib/aws-secrets-manager/SecretsManagerStack.ts:19]: That makes sense.

We might want a different test which verifies secrets manager independently of OpenSearch. In that case, we'd want a more unique username/password.

**Review by dlvenable — APPROVED** [review]

**Review by oeyh — APPROVED** [review]

**Pull request merged**